upup/pkg/fi/cloudup/gce/gce_cloud.go

```diff
@@ -29,24 +30,41 @@ import (
 	"k8s.io/kubernetes/federation/pkg/dnsprovider/providers/google/clouddns"
 )
 
-type GCECloud struct {
-	Compute *compute.Service
-	Storage *storage.Service
+type GCECloud interface {
```

Comment:
Not sure if it is a lot of work, but yah I was thinking about that this afternoon.  Now if I just understood interfaces better :P